lib/endpoints/commit.js

```diff
@@ -5,19 +5,68 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 var restify = require('restify');
-var errors = require('../errors');
 
+var common = require('../common');
+var errors = require('../errors');
 
 
 /**
  * POST /commit
+ *
+ * Note: These fields are already populated:
+ *   req.params.id (unescaped container id/name)
+ *   req.vm (container vmObj)
  */
 function commit(req, res, next) {
-    return next(new errors.NotImplementedError('commit'));
+    var log = req.log;
+
+    // Create name string in format 'repo:tag'
+    var repoAndTag = req.query.repo || '';
+    if (req.query.tag) {
+        repoAndTag += ':' + req.query.tag;
+    }
+
+    // There can either be one single change (string) or multiple (array of
+    // string) changes - convert to always be an array.
+    var changes = req.query.changes || [];
+    if (changes && typeof (changes) === 'string') {
+        changes = [changes];
+    }
+
+    var commitOpts = {
+        author: req.query.author,
+        changes: changes,
+        comment: req.query.comment,
+        config: req.body,
+        pause: common.boolFromQueryParam(req.query.pause),
+        tag: repoAndTag
+    };
+
+    log.debug('commit: req.query: ', req.query);
+
+    /*
+     * Node's default HTTP timeout is two minutes, and the client context
+     * request can take longer than that to complete.  Set this connection's
+     * timeout to an hour to avoid an abrupt close after two minutes.
+     */
+    req.connection.setTimeout(60 * 60 * 1000);
+
+    req.backend.commitImage(req, commitOpts, function (err, imageId) {
+        if (err) {
+            res.send(err);
+        } else {
+            log.debug('docker commit finished successfully');
+            // Note: Docker returns sha256: {"Id":"sha256:93c3e0ca32...1453a"}
+            res.write(JSON.stringify({'Id': imageId}) + '\n');
+        }
+
+        res.end();
+        next(true);
```

Comment:
I'll change that to *next()*. I had thought it meant that we are done with the request and there are no more handlers to run for this restify route, but reading the restify docs and code I find it confusing.